Documentation/2.1-update-process.md

```diff
@@ -0,0 +1,62 @@
+# Process for Updating source-build for 2.1
+
+## Source Build Repo Structure
+
+The source-build repo pulls together all of the repos used to build the dotnet core product and builds them in succession to establish a baseline of packages needed.  It then packages all source into a tarball that can be used by partner teams to build from source.
+
+The following are the key source-build sub-directories:
+
+* patches - Contains a directory for each repo that requires patches to its source to build correctly.
+* repos - Contains projects, props and targets to orchestrate the build, including one project for each repo.  \<RepositoryReference> tags are used to identify dependencies between repositories.  The cli project is the top-level project, which rolls up all others.
+* src - Contains a submodule for each repo.
+
+## Build Stages
+
+The source-build process consists of 3 main stages:
+
+1. **Build source-build repo**
+
+    * Install tools (CLI and BuildTools)
+    * Apply temporary patches to repos as applicable
+    * Build each repo
+
+1. **Generate tarball**
+    * Copy Tools to tarball
+    * Copy sources to tarball
+    * Create prebuilt packages directory
+        * Copy each restored package to the prebuilt directory
+        * Remove any packages generated as part of the source-build build
+
+1. **Build tarball in an offline environment**
+
+## Auto-dependency flow API
+
+Source-build defines a set of properties to assist in building and placing output.  These are:
+
+* `DotNetPackageVersionPropsPath` - To identify the versions of built pacakges to depend on.
+* `DotNetBuildOffline` - Identifies that build is indended to be offline.
+* `DotNetRestoreSourcePropsPath` - Where to restore packages from during build.
+* `DotNetOutputBlobFeedDir` - Where to place output during build.
```

Comment:
We probably want to reconcile this with the similar property we have with product con.

Reply:
Which property?  The prodcon property for publish is a URL to publish to, while this one is just a directory.

Reply:
The URL could also be a local directory I believe the publish task supports both. At any case even if they don't currently align we need to make sure we reconcile them to avoid unnecessary overlap.